Added WIZnet IC's

NewFork17 opened the pull request:
**Notes:**
- Kept W5100 as original as possible. Minor changes in meta data (according datasheet).
- Added all other WIZnet IC's to the lib (All WIZnet IC's except W5100)
- Followed KiCAD guidelines.
- Kept presentation "simular" to W5100.
- Fixed CLAassistant remarks
- Some datasheet combine 2 different footprints (with different pin numbers) where the chip itself is not quit the same. Kept the naming as a lot of salespoints and Wiznet uses theirself by using _q or _l as extension for QFN or LQFP.

**URL's:**
* [W3150A](http://wiznethome.cafe24.com/wp-content/uploads/wiznethome/Chip/W3150A_Plus/Documents/W3150A+_DS_V206E.pdf)
* [W5100](https://www.wiznet.io/wp-content/uploads/wiznethome/Chip/W5100/Document/W5100_DS_V128E.pdf)
* [W5100S](https://wizwiki.net/wiki/lib/exe/fetch.php?media=products:w5100s:w5100s_ds_v125e.pdf)
* [W5200](http://wiznethome.cafe24.com/wp-content/uploads/wiznethome/Chip/W5200/Documents/W5200_DS_V140E.pdf)
* [W5300](https://www.wiznet.io/wp-content/uploads/wiznethome/Chip/W5300/Documents/W5300_DS_V134E.pdf)
* [W5500](http://wizwiki.net/wiki/lib/exe/fetch.php/products:w5500:w5500_ds_v109e.pdf)
* [W6100](https://wizwiki.net/wiki/lib/exe/fetch.php/products:w6100:w6100_ds_v104e.pdf)
* [W7100A](https://www.wiznet.io/wp-content/uploads/wiznethome/Chip/W7100A/Document/W7100A_DS_V124E.pdf)
* [W7500X](https://wizwiki.net/wiki/lib/exe/fetch.php?media=products:w7500:w7500x_ds_v110.pdf)

################

All contributions to the kicad library must follow the KiCad library convention

Thanks for creating a pull request to contribute to the KiCad libraries! To speed up integration of your PR, please check the following items:

- [x]  Provide a URL to a datasheet for the symbol(s) you are contributing
- [x]  Provide a screenshot of the symbol(s) from the symbol editor with the pin types visible
- [x]  Ensure that the associated footprints match the official footprint library
- [x] A new fitting footprint must be submitted if the library does not yet contain one. => N/A
If there are matching footprint PRs, provide link(s) as appropriate
- [x]  Check the output of the Travis automated check scripts - fix any errors as required
- [x]  Give a reason behind any intentional library convention rule violation. => N/A
- [x] Be patient, we maintainers are volunteers with limited time and need to check your

contribution against the datasheet. You can speed up the process by providing all the necessary information (see above). And you can speed up the process even more by providing additional info like the screenshot of the symbol editor pin table (or for high pin counts converted to csv) sorted in the same way as the pin table in the datasheet and a direct link to the datasheet page that contains the pin table.

################

**Screenshots:**
![W3150A](https://user-images.githubusercontent.com/54643555/84934461-a6232600-b0d7-11ea-8e41-320f38f26bb3.png)
![W5100](https://user-images.githubusercontent.com/54643555/84934465-a6bbbc80-b0d7-11ea-934a-487361d67bec.png)
![W5100S_L](https://user-images.githubusercontent.com/54643555/84934467-a7545300-b0d7-11ea-9c49-bdd2b406f888.png)
![W5100S_Q](https://user-images.githubusercontent.com/54643555/84934470-a7545300-b0d7-11ea-92fd-dcc29017b0cc.png)
![W5200](https://user-images.githubusercontent.com/54643555/84934471-a7ece980-b0d7-11ea-8a81-d36d8e453bdd.png)
![W5300](https://user-images.githubusercontent.com/54643555/84935455-0797c480-b0d9-11ea-912b-080949153927.png)
![W5500](https://user-images.githubusercontent.com/54643555/84935459-08305b00-b0d9-11ea-86d2-405644b822f7.png)
![W6100_L](https://user-images.githubusercontent.com/54643555/84934475-a8858000-b0d7-11ea-913d-d74c74bb4983.png)
![W6100_Q](https://user-images.githubusercontent.com/54643555/84934476-a8858000-b0d7-11ea-83ea-ec17ad4e8480.png)
![W7100A_L](https://user-images.githubusercontent.com/54643555/84934480-a91e1680-b0d7-11ea-9f49-4271d3f0d303.png)
![W7100A_Q](https://user-images.githubusercontent.com/54643555/84934481-a91e1680-b0d7-11ea-8b4c-fbcc9eae8a7a.png)
![W7500](https://user-images.githubusercontent.com/54643555/84934484-a9b6ad00-b0d7-11ea-9351-e9c789b1ebbc.png)
![W7500P](https://user-images.githubusercontent.com/54643555/84934486-a9b6ad00-b0d7-11ea-8fd6-4e5d4f82ed4e.png)

klepklep17 commented:
**Picture of all items added after succesful pulls and checks.**

![W3150A](https://user-images.githubusercontent.com/54643555/83335375-bac88700-a2ac-11ea-941d-62ea9f588636.png)
![W5100](https://user-images.githubusercontent.com/54643555/83335376-bb611d80-a2ac-11ea-82cf-dd4d025adf8e.png)
![W5100S_L](https://user-images.githubusercontent.com/54643555/83335377-bb611d80-a2ac-11ea-9c2b-6f2385d206ef.png)
![W5100S_Q](https://user-images.githubusercontent.com/54643555/83335378-bbf9b400-a2ac-11ea-85be-5586484d98cd.png)
![W5200](https://user-images.githubusercontent.com/54643555/83335379-bbf9b400-a2ac-11ea-893a-e4358f3d8008.png)
![W5300](https://user-images.githubusercontent.com/54643555/83335380-bbf9b400-a2ac-11ea-8ca6-115481a010b9.png)
![W5500](https://user-images.githubusercontent.com/54643555/83335381-bc924a80-a2ac-11ea-99e1-23ea9de20ec5.png)
![W6100_L](https://user-images.githubusercontent.com/54643555/83335382-bc924a80-a2ac-11ea-8757-7baae1c72271.png)
![W6100_Q](https://user-images.githubusercontent.com/54643555/83335383-bc924a80-a2ac-11ea-8585-fe6b2c4e0e54.png)
![W7100A_L](https://user-images.githubusercontent.com/54643555/83335384-bd2ae100-a2ac-11ea-97e6-14b5d8865f0e.png)
![W7100A_Q](https://user-images.githubusercontent.com/54643555/83335385-bd2ae100-a2ac-11ea-99cc-48ad4315f404.png)
![W7500](https://user-images.githubusercontent.com/54643555/83335386-bd2ae100-a2ac-11ea-92f9-5cfa92782e44.png)
![W7500P](https://user-images.githubusercontent.com/54643555/83335387-bdc37780-a2ac-11ea-9367-06c9f1d6d13d.png)

aewallin commented:
- package name at end of description?
- 'Mb' for megabits per second is a bit of a shorthand... maybe 'Mbit/s'

NewFork17 commented:
Thx aewallin for your response.

Reaction on your input:
* Package name is not needed due to the unique name of every package variation. Also there are no other brands supplying these parts due to copyrights/patents. As I understand is conform with KLC S2.1.
* The Mb parts is short. I did not change the original naming of the part W5100. So it is a copy of the original library part. I will change that. Your are right to make that the right way.

chschlue commented:
@NewFork17 please take a look at [S6.2](https://kicad-pcb.org/libraries/klc/S6.2/), package name in the description is always required if the symbol is fully-specified. S2.1 does not deal wiith metadata.

NewFork17 commented:
@chschlue : Tnx. I did not know. I don't understand why exactly this is needed, but I will add this.

NewFork17 commented:
Updated according feedback KLC S6.2

NewFork17 commented:
@aewallin  & @chschlue : S6.2 Done
I still have to learn how to combine all new things of making, donating, and working via GitHub. Thx for your patients.

aris-kimi commented:
Hi @NewFork17, i had the same problems... and still have!! If you open a pull request(PR), and you want to change something, you go back to YOUR branch from where the PR was created, you upload the files you've changed, and just commit changes.
If YOUR branch is some commits behind kicad master branch you should syncronize it first.
Hope this info helped.

NewFork17 commented:
@aris-kimi : Tnx. That confirms what I think I should do. Especially to update the files in the branch. If I update/pull it via github-desktop-app after a PR  I am missing something.

NewFork17 commented:
@Anyone: There are 2 labels "Addition" and "Pending reviewer".  What is next? Leave it here unit ....? Can I close this branch? Is it copied? May I rename it? Do I have to do or is my "contribution" finished?

chschlue commented:
Wait for the PR to get reviewed.
It definitely helps attracting volunteers to review it if all necessary information can be found right on top of the page, which is why we have the [PR template](https://github.com/KiCad/kicad-symbols/blob/master/.github/PULL_REQUEST_TEMPLATE.md) that you could (and should) have followed instead of overwriting it.

NewFork17 commented:
@chschlue: I am missing some point. As far as I understand I have followed the KiCad contribution guidelines, the KLC's and the mentioned PR template. You are writing "could and should" and "following instead of overwriting". There was/is some learning curve for me. What have I missed?

I am not ready to give an official approval, however if required I can review additions (at least some steps) for other pull requests than me. How does that work?

chschlue commented:
From your initial post:
> Kept W5100 as original.

You didn't.

From the PR template:
> Provide a URL to a datasheet for the symbol(s) you are contributing

You didn't.

---------

All librarians are volunteers reviewing contributions in their spare time, which is limited.
By not adding the necessary details to the topmost post (especially screenshots and their respective datasheets) so potential reviewers can't even start without scrolling and clicking around you create extra work for no reason.

I can only speak for myself here, but I for one tend to just skip weirdly formatted PRs that kill my workflow.
What exactly is the problem with doing as the Romans do (unless you have important reasons to deviate)?

NewFork17 commented:
@chschlue:

I really balancing my options now. I though I provided a piece of contribution so I consider myself also a (small) volunteer by added a set of new components where I actually need just one. It consumed me 2 weekends (sparetime) to learn KiCad way of contributing and learning GitHub and of course building the symbols. There is even a PR with lots of valves/tubes coming "your" way.

There is information what is required in KLC's. Only there is very little explanation about how the process of Github and the KiCad volunteering team is working. So it is very easy to do wrong.

Reaction on W5100: I got a comment in this conversation about meta data not being okay so I updated. Should I have not done that? Or should I made a new contribution? Or should I made this clear in a certain way?

Reaction on  about datasheets: Everything I added had a datasheet reference. So that confusing me. Should I also add URL's in the PR? Because there are URL's in the symbol library (dcm).

I added screenshots. Not the first post, got to learn, made a "mistake".

I really need some good input here because otherwise I will drop this ball. Github is not easy. And I understand your position like when I am explaining my parents how they can use the laptop/smartphone for the 100 times or so. I am not stupid, not even close. Everybody needs to start somewhere. I will make mistakes, due to the reason I am human. 

Status of first post, doing it Romans way:
* Provide a URL to a datasheet for the symbol(s) you are contributing => Done, is in KiCad dcm-file
* Provide a screenshot of the symbol(s) from the symbol editor with the pin types visible => Done, not the first post. (Learning)
* Ensure that the associated footprints match the official footprint library => Done
* If there are matching footprint PRs, provide link(s) as appropriate => Not applicable
* Check the output of the Travis automated check scripts - fix any errors as required => Learning process of KLC, Github, CLAassistant and symbol editor.
* Give a reason behind any intentional library convention rule violation. => No Need

aris-kimi commented:
Hi @NewFork17 check some other symbol PRs and you will spot the differences, in a PR you provide a datasheet link and a screenshot of the symbol. I haven't use github desktop at all, i only use browser interface, and my tactic is, one branch for every new symbol and one PR, if there are other aliases for THIS particular symbol, i upload changed files at the current branch, after i syncronize it, and the PR gets updated automaticaly. Those labels are nothing but description for the state of the PR so it can make librarian's life comfortable.. When a PR gets to open, it is waiting for a librarian to review the changes, when the job gets done, and the changes are merged, then you can safely delete your branch with a click at the end of the PR.. Hope i helped... We are all humans by the way, and we all spend our free time for this.

NewFork17 commented:
@aris-kimi 
Tnx. This is a lot of good info. I'll think we (in general) need a kind of make this way of work a part of the kicad website or GitHub or something. Set an example. I experience the given info to much stacato. Better info provides better workflow for people like @chschlue.

@chschlue 
Also tnx up till now. I appreciate your effort very much.

chschlue commented:
> Reaction on W5100: I got a comment in this conversation about meta data not being okay so I updated. Should I have not done that? Or should I made a new contribution? Or should I made this clear in a certain way?

The problem is not the metadata, but that you moved the GNDA pins. This makes the whole PR a breaking change which has to be deferred to v6.

> Reaction on about datasheets: Everything I added had a datasheet reference. So that confusing me. Should I also add URL's in the PR? Because there are URL's in the symbol library (dcm).

Yes, please. This enables reviewers to take a quick look what the PR is about without either pulling the branch or sifting through the and manually copying the URLs.

> I really need some good input here because otherwise I will drop this ball. Github is not easy. And I understand your position like when I am explaining my parents how they can use the laptop/smartphone for the 100 times or so. I am not stupid, not even close. Everybody needs to start somewhere. I will make mistakes, due to the reason I am human.

I get that. So am I. No hard feelings.

> Status of first post, doing it Romans way:

Please actually update the first post so the next guy that comes around and might have time on his hands to review this rather extensive PR isn't put off by having to read through this entire thread first.

NewFork17 commented:
@chschlue Little by little I get what you experience. Tnx.

I was not aware of this change W5100.

I will change first post!!!

Schlumpf commented:
Hi,
I'm not a librarian, but did a quick review. Here are some points to change or discuss:

1. you should increase the distance between the crystal pins so that you can draw a crystal symbol in between. Here is an example from W5100:
![grafik](https://user-images.githubusercontent.com/329423/84566092-d03eb600-ad6e-11ea-961d-e88f72612dff.png)
It is quite common throughout the library. For MCUs, too.

2. Another question is where to place the voltage output pin. W5100 has it it on top, the other symbols on the right. We should keep it consistent. Maybe @chschlue has an option here?

3. I checked the reference schematics of all parts for the LED pins. In my opinion they should be all open collector, even if the datasheet tells "output". And they should all be marked as negated.

4. The RESET_BG pin needs usually two resistors against GND. You put it at the bottom right. If there is enough space on the side, you could lift the pin a little bit so that the resistors do not stick out under the symbol. That would save some space on the page. Example from W5100:
![grafik](https://user-images.githubusercontent.com/329423/84567098-adb09b00-ad76-11ea-86c9-7d4cca6c29d6.png)

5. The RX/TX pins on the right are two groups at W5100 and one group at the other parts. I think this should be consistent, too.

6. For the symbols W5100S_L, W5100S_Q, W5200, W6100_L and W6100_Q the left side is relatively full and there is still space on the right side. To get the symbols a bit smaller, you could move the crystal pins to the right.

7. You could reduce the height of W5500 by moving the NC pins between the regulation pins.

best regards
Martin

NewFork17 commented:
@Schlumpf,

Tnx for your reaction. Quit a feedback, comment and input.

Ad1:
This is a personal opinion what is correct or the best. I assume that your input is correct that many MCU have space.  For this I do like to make this compact and put the crystal anywhere else on the schematic due that it is a "utility" and it does not contribute much to the understanding of a MCU in the design. For me the W5100 is not completely according general schematic expectations.

General schematic expectations: You showed me an example of the "right-side". This is also for me not done.  In my opinion schematic should/must mainly be readable from right to left and voltages/current go from top to bottom, with the exception of things like feedback signals. For large objects like these I divide the bidirectional inputs on left and right side of the object. Only the power supply should be on top or bottom depending on voltage levels.

So for me you put an interesting topic in this conversation. I am really curious about other opinions.

Ad2:
Output voltages should be on the right side for me. That is why the LAN connection and the leds are on the right side.

Ad3:
I am going to look into this.

Ad4:
2 resistor in series does not make any technical sense, so I going to dig into this whats the reason for that. Tnx for the info/tip. For the position of the pin: that can be improved!

Ad5:
Okay.

Ad6:
See Ad1 about reading / flow from left to right. In my opinion that's is less readable.

My opinion: A electronic result is about the idea, concept, design that is rather easily readable in the schematics and PCB layout. For me several more or bigger pages is not a problem as long as the schematics can be read like a book. If I have to make a study of a schematic, the original thoughts and choices are lost. For me that is general schematic expectation where we have to make the tools around, not vica versa.

So I am really curious what the opinions are of others KiCad users.

Ad7:
Do you think I should stack the NC's to reduce space?

Ad8:
I understand your feedback on this. I can change this back.

Than I have a question: what to do if things can be improved (not the same as error correction), but will lead to none compatible? Does that mean we can't improve anymore ones a design has been put in the library?

Tnx for your reaction. Appreciate it.

chschlue commented:
I've been mentioned so many times here and I'm hoping I'll finally be able to get back to this during the weekend...

Just one little comment regarding RSET_BG for now:
I assume the 12k + 0.3k recommendation stems from the fact that a 12.3k resistor is E192 and is hard to come by, paired with tolerance overzealousness (mind that this isn't supposed to be temperature controlled or anything). Pretty sure a single E96 12.4k is fine, too.

Schlumpf commented:
W5500 has been merged at https://github.com/KiCad/kicad-symbols/pull/2425. We should remove it from this PR.
